src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/CourtService.kt

```diff
@@ -236,6 +242,33 @@ class CourtService(
       )
     }
 
+  private fun validateVideoLinkBookingSpecification(specification: VideoLinkBookingSpecification) {
+    validateVideoLinkAppointment(specification.main, "Main")
+    specification.pre?.let { validateVideoLinkAppointment(it, "Pre") }
+    specification.post?.let { validateVideoLinkAppointment(it, "Post") }
+  }
+
+  private fun validateVideoLinkBookingUpdateSpecification(specification: VideoLinkBookingUpdateSpecification) {
+    validateVideoLinkAppointment(specification.main, "Main")
+    specification.pre?.let { validateVideoLinkAppointment(it, "Pre") }
+    specification.post?.let { validateVideoLinkAppointment(it, "Post") }
+  }
+
+  private fun validateVideoLinkAppointment(appointment: VideoLinkAppointmentSpecification, prefix: String) {
+    if (appointment.startTime.isBefore(LocalDateTime.now(clock))) {
+      throw ValidationException("$prefix appointment start time must be in the future.")
+    }
+    if (!appointment.startTime.isBefore(appointment.endTime)) {
+      throw ValidationException("$prefix appointment start time must precede end time.")
+    }
+    appointment.locationId?.let {
+      val location = prisonApiService.getLocation(it) ?: throw ValidationException("$prefix locationId $it not found in NOMIS.")
+      if (VIDEO_LINK_APPOINTMENT_LOCATION_TYPE != location.locationType) {
```

Comment:
can we do this? the prison side doesn't limit the location list to just video link appointments